sdk/schemaregistry/schema-registry-json/src/jsonSerializer.ts

```diff
@@ -190,7 +229,7 @@ function convertMessage<MessageT>(
 }
 
 function getSchemaName(schema: SchemaObject): string {
-  const id = schema.$id || schema.id;
+  const id = schema.$id;
```

Comment:
is this change necessary?

Reply:
From what I see, a schema only needs `$id` but not `id`, I'm not sure if a schema has both.